src/components/PartySwitch/PartySwitch.tsx

```diff
@@ -97,6 +102,29 @@ export const PartySwitch = ({
     [selectedId]
   ) as PartySwitchItem;
 
+  useEffect(() => {
+    setVisibleParties(filteredParties.slice(0, 50));
+  }, [filteredParties]);
```

Comment:
`useEffect` should not be used to sync states.
We should derive the visible parties information from `filteredParties` plus an offset state.

My proposal is:
- Remove the `visibleParties` state and replace it with an `visiblePartiesOffset` that contains a number, initialize it to 50;
- In the `loadMoreParties` function, increase the `visiblePartiesOffset` state by the desired amount;
- During the render derive the `visibleParties` information from the `filteredParties` and the `offsetState`. (`const visibleParties = filteredParties.slice(0, visiblePartiesOffset)`)

LMK if this makes sense to you.

Reply:
Thank you for your feedback! Your proposal makes sense, and I've implemented the changes accordingly.

I've replaced the visibleParties state with an offset state, which tracks the number of visible parties. 
Additionally, I've updated the loadMoreParties function to increase the offset state when loading more parties.
Now, during rendering, I'm deriving the visibleParties information directly from filteredParties and the offset state using filteredParties.slice(0, offset).